src/openrct2/peep/Guest.cpp

```diff
@@ -4179,51 +4179,63 @@ void Guest::UpdateRideEnterVehicle()
 {
     Ride* ride = get_ride(current_ride);
 
-    rct_vehicle* vehicle = GET_VEHICLE(ride->vehicles[current_train]);
-    for (int32_t i = current_car; i != 0; --i)
+    if (ride != nullptr)
     {
-        vehicle = GET_VEHICLE(vehicle->next_vehicle_on_train);
-    }
+        rct_vehicle* vehicle = GET_VEHICLE(ride->vehicles[current_train]);
 
-    if (ride->mode != RIDE_MODE_FORWARD_ROTATION && ride->mode != RIDE_MODE_BACKWARD_ROTATION)
-    {
-        if (current_seat != vehicle->num_peeps)
-            return;
-    }
+        if (vehicle != nullptr)
+        {
+            for (int32_t i = current_car; i != 0; --i)
+            {
+                vehicle = GET_VEHICLE(vehicle->next_vehicle_on_train);
+            }
 
-    if (vehicle_is_used_in_pairs(vehicle))
-    {
-        auto seated_peep = (GET_PEEP(vehicle->peep[current_seat ^ 1]))->AsGuest();
-        if (seated_peep == nullptr || seated_peep->sub_state != PEEP_RIDE_ENTER_VEHICLE)
-            return;
+            if (ride->mode != RIDE_MODE_FORWARD_ROTATION && ride->mode != RIDE_MODE_BACKWARD_ROTATION)
+            {
+                if (current_seat != vehicle->num_peeps)
+                    return;
+            }
 
-        vehicle->num_peeps++;
-        ride->cur_num_customers++;
+            if (vehicle_is_used_in_pairs(vehicle))
+            {
+                auto* seated_peep = GET_PEEP(vehicle->peep[current_seat ^ 1]);
```

Comment:
Since you are touching most of this function may as well switch to camelCase for all of the variables.

Reply:
OK, I changed all the local function variables...if you want to create new issues to refactor some of the other var names in the Peep and Guest structs I can take a crack at those also.

Reply:
OK I updated the vars that were local to the function. If you want to open some new issues to refactor the names in `Peep` and `Guest` structs I can take a crack at those, too.